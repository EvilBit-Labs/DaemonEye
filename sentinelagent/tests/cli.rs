use assert_cmd::prelude::*;
use predicates::prelude::*;
use std::process::Command;

#[test]
fn prints_expected_greeting() -> Result<(), Box<dyn std::error::Error>> {
    use tempfile::tempdir;

    let temp_dir = tempdir()?;
    let db_path = temp_dir.path().join("test.db");

    let mut cmd = Command::cargo_bin("sentinelagent")?;
    cmd.env("SENTINELAGENT_DATABASE_PATH", db_path.to_str().unwrap());
    // Enable test mode so the agent exits immediately after startup banner.
    cmd.env("SENTINELAGENT_TEST_MODE", "1");
    cmd.assert().success().stdout(predicate::str::contains(
        "sentinelagent started successfully",
    ));
    Ok(())
}
