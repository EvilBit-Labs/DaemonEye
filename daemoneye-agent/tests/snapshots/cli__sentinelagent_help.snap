---
source: sentinelagent/tests/cli.rs
expression: stdout
---
SentinelD Detection and Alerting Orchestrator

Usage: sentinelagent [OPTIONS]

Options:
  -d, --database <DATABASE>    Database path [default: /var/lib/sentineld/processes.db]
  -l, --log-level <LOG_LEVEL>  Log level [default: info]
  -h, --help                   Print help
  -V, --version                Print version
